### ISSUE TITLE
Native crash caused by JS in renderer (advert?)

### ISSUE THREAD

**Issue by Niquolas** [electron/electron#23233]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  * 9.0.0-beta.18 (also reproducible in 8.2.3)
* **Operating System:**
  * Windows 10 1909
* **Last Known Working Electron version:**
  * Unknown

### Expected Behavior
The web page to behave as expected.

### Actual Behavior
Electron crashes:

```
Received fatal exception EXCEPTION_ACCESS_VIOLATION

Backtrace:

(No symbol) [0x0000000000000000]

std::__1::vector<v8::CpuProfileDeoptInfo,std::__1::allocator<v8::CpuProfileDeoptInfo> >::get_allocator [0x00007FF6BF946448+58424]

uv_fs_get_statbuf [0x00007FF6C15FB25C+86348]

uv_cond_signal [0x00007FF6C1ADB90E+938254]

uv_cond_signal [0x00007FF6C1AE018D+956813]

uv_cond_signal [0x00007FF6C1ADFD87+955783]

uv_cond_signal [0x00007FF6C1ADA320+932640]

uv_cond_signal [0x00007FF6C1ADABE7+934887]

uv_cond_signal [0x00007FF6C1AEBCD0+1004752]

uv_mutex_unlock [0x00007FF6C19B67B1+2152321]

uv_gettimeofday [0x00007FF6C21774A9+225113]

uv_gettimeofday [0x00007FF6C2177230+224480]

uv_mutex_unlock [0x00007FF6C19F2374+2396996]

uv_mutex_unlock [0x00007FF6C19F1ADE+2394798]

uv_gettimeofday [0x00007FF6C2177AB8+226664]

uv_mutex_unlock [0x00007FF6C19A239A+2069354]

v8_inspector::V8StackTraceId::ToString [0x00007FF6C103D772+5951442]

v8_inspector::V8StackTraceId::ToString [0x00007FF6C103D643+5951139]

v8_inspector::V8StackTraceId::ToString [0x00007FF6C103F2C1+5958433]

v8_inspector::V8StackTraceId::ToString [0x00007FF6C103AFED+5941325]

v8_inspector::V8StackTraceId::ToString [0x00007FF6C0F4022D+4913805]

v8_inspector::V8StackTraceId::ToString [0x00007FF6C0F40DD1+4916785]

v8_inspector::V8StackTraceId::ToString [0x00007FF6C0F40A93+4915955]

IsSandboxedProcess [0x00007FF6C1EA8BC3+310707]

v8::ResourceConstraints::ResourceConstraints [0x00007FF6C0314F3E+1398462]

Ordinal0 [0x00007FF6BF87140B+5131]

node::AsyncResource::CallbackScope::CallbackScope [0x00007FF6C4CCB982+21725986]

BaseThreadInitThunk [0x00007FFF6E917BD4+20]

RtlUserThreadStart [0x00007FFF6F2ACE51+33]

Electron exited with code 3221225477.
```

### To Reproduce
See electron fiddle at:

https://gist.github.com/Niquolas/81cc70179578975651e7efd94d6f9748

Ensure any adblockers e.g. Pi-hole are disabled.

1) Launch the application
2) Wait for the page to fully load
3) Interact with the page (clicking on an ad yields the most consistent results)

### Additional Information
There seems to be a script on the page (probably an advert, as enabling Pi-hole negates the issue for me) that is causing the native crash.

Not sure if this is an Electron bug or something further upstream, so I thought it best to check here. Apologies if this is out of scope.

**Comment by ckerr** [comment]
(FWIW I'm not able to reproduce this issue when testing on Linux)

**Comment by ckerr** [comment]
This *may* be a duplicate of #22513 which has a pending fix. @Niquolas you may want to keep an eye on that issue and try a nightly when that fix lands

**Comment by deepak1556** [comment]
Thanks for the repro, was able to verify the crash, this definitely has the same cause as https://github.com/electron/electron/issues/22513 . Please follow that issue for updates and also do try the night builds once the PR is merged as suggested by @ckerr
